sae_lens/config.py

```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
+        model_batch_size (int): How many prompts are in the batch of the language model when generating activations.
+        hook_name (str): The name of the hook to use.
+        final_hook_layer (int): The layer of the final hook. Currently only support a single hook, so this should be the same as hook_name.
+        d_in (int): Dimension of the model.
+        total_training_tokens (int): Total number of tokens to process.
+        context_size (int): Context size to process. Can be left as -1 if the dataset is tokenized.
+        model_class_name (str): The name of the class of the model to use. This should be either `HookedTransformer` or `HookedMamba`.
+        activation_save_path (str, optional): The path to save the activations.
+        shuffle (bool): Whether to shuffle the dataset.
+        seed (int): The seed to use for shuffling.
+        dtype (str): Datatype of activations to be stored.
+        device (str): The device for the model.
+        buffer_size_gb (float): The buffer size in GB. This should be < 2GB.
+        hf_repo_id (str, optional): The Hugging Face repository id to save the activations to.
+        hf_num_shards (int, optional): The number of shards to save the activations to.
+        hf_revision (str): The revision to save the activations to.
+        hf_is_private_repo (bool): Whether the Hugging Face repository is private.
+        model_kwargs (dict): Keyword arguments for `model.run_with_cache`.
+        model_from_pretrained_kwargs (dict): Keyword arguments for the `from_pretrained` method of the model.
+        compile_llm (bool): Whether to compile the LLM.
+        llm_compilation_mode (str): The torch.compile mode to use.
+        prepend_bos (bool): Whether to prepend the beginning of sequence token. You should use whatever the model was trained with.
+        seqpos_slice (tuple): Determines slicing of activations when constructing batches during training. The slice should be (start_pos, end_pos, optional[step_size]), e.g. for Othello we sometimes use (5, -5). Note, step_size > 0.
+        streaming (bool): Whether to stream the dataset. Streaming large datasets is usually practical.
+        autocast_lm (bool): Whether to use autocast during activation fetching.
+        dataset_trust_remote_code (bool): Whether to trust remote code when loading datasets from Huggingface.
     """
 
-    # Data Generating Function (Model + Training Distibuion)
-    model_name: str = "gelu-2l"
-    model_class_name: str = "HookedTransformer"
-    hook_name: str = "blocks.{layer}.hook_mlp_out"
-    hook_layer: int = 0
-    hook_head_index: Optional[int] = None
-    dataset_path: str = ""
-    dataset_trust_remote_code: bool | None = None
-    streaming: bool = True
-    is_dataset_tokenized: bool = True
-    context_size: int = 128
-    new_cached_activations_path: Optional[str] = (
-        None  # Defaults to "activations/{dataset}/{model}/{full_hook_name}_{hook_head_index}"
-    )
+    hf_dataset_path: str
```

Comment:
this looks like a breaking change. Do we need to change this name? This would require a major version bump and would be annoying for anyone currently using the library as they will need to change their configs